### PR TITLE
Damage conservation on set_species, i.e. changeling transform no longer heals everything.

### LINKED ISSUE #8088: Changeling Instant Regenerate
**Problem Description**:
Changelings can instantly regenerate all damage and slip out of handcuffs (and drop some other clothing too, can't remember off the top of my head what else) by transforming into a species different from their current species.

**What did you expect to happen**: 
I expect to retain all damage and clothing/cuffs when transforming.

**What happened instead**:
I was able to live nearly forever by constantly changing into a different species, slipping out of cuffs that were put on me and removing ALL inflicted damages, be the damage external or internal.

**Why is this bad/What are the consequences:**
It's a very easily abused exploit that allows for near invincibility in changelings.

**Steps to reproduce the problem**:
Be a changeling, absorb DNA of a species that isn't your current one, take any amount of damage, transform into different species, observe that you have been fully healed and dropped some clothes including cuffs.

**Possibly related stuff (which gamemode was it? What were you doing at the time? Was
anything else out of the ordinary happening?)**: I was able to reproduce this in a couple shifts and have seen others do it.

### PR DESCRIPTION
**Pull request by Birdtalon**
This PR adds a check to `set_species()` for conservation of damage to bodyparts and organs when the species change is done.
It's been a long known and used behaviour that changing species using changeling transform will fully heal you, repair all bones, organs etc. This PR attempts to address that by conserving damage primarily on changeling transform, but it can be applied anywhere that `set_species()` is used.

The PR checks all bodyparts (external organs) and internal organs for damage vars, as well as internal bleeding in the case of external organs. Saves any which have damage and discards the balance. The organs are then deleted when the new species organs are created. We then loop through the new organs and check the saved ones against them. When a match is found; damage is applied and the saved values are then discarded. Broken internal organs are carried.

If `retain_damage = TRUE` is not provided to `set_species()` all of this is ignored as before.

Edge cases:

- Organs which do not exist in the species being transformed to never find a match and damage is therefore not applied as there is no equivalent organ. For example, the Vox  race has the stack organ - which means that the Vox could technically transform to a Human, and back to a Vox and the organ damage will not be conserved.

fixes #8088

🆑 Birdtalon
add: Conservation of damage mechanic for set species
tweak: Changeling transform conserves damage when transforming to a different species.
/🆑

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1325,7 +1325,63 @@
...
+					E.receive_damage(brute, burn, ignore_resists = TRUE)
+					E.internal_bleeding = IB
+					qdel(part)
+
+		for(var/obj/item/organ/internal/I in internal_organs)
+			for(var/list/part in internal_damages)
+				var/obj/item/organ/internal/OI = part[1]
+				var/organ_type
+
+				if(OI.parent_type == /obj/item/organ/internal) //Dealing with species organs
```
Comment (Birdtalon) [inline code/modules/mob/living/carbon/human/human.dm]: Species organs are sibling classes, I wasn't sure if there was already a way to check this.

**Comment by Aurorablade** [comment]
would it not be better to split it all out into its own proc instead of all on set_species but called IN set_speices, then set a flag/var on ling trans(or other transforms) to trigger the check if its true/set?

**Comment by Birdtalon** [comment]
Because the deletion and re-creation of organs is handled in `create_organs()` and the necessity to save these values before the organs are deleted. The only other logical place to do it is within `create_organs()` itself and have a parameter on both procs which would allow for breaking it up further.

I don't think that creating a new proc is necessary in this case as all I would be doing is returning certain states of bodyparts and organs which I can't see being used in any other way. If you see merit in breaking it up into separate procs then I'm open to suggestion.

**Comment by Birdtalon** [comment]
Added handling for broken internal organs.

**Pull request merged**